skill-tree /src/main/java/com/RDS/skilltree/utils/JWTUtils.java

```diff
@@ -0,0 +1,86 @@
+package com.RDS.skilltree.utils;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.AuthenticationCredentialsNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Component
+@Slf4j
+public class JWTUtils {
+
+    @Value("${jwt.rds.public.key}")
+    private String publicRDSKeyString;
+    private static ConcurrentHashMap<String, RSAPublicKey> publicKeyCache = new ConcurrentHashMap<>();
+
+    /**
+     * Converts the given private key string to an RSAPrivateKey object.
+     *
+     * @param privateKeyString The private key string to be converted.
+     * @return The corresponding RSAPrivateKey object.
+     * @throws Exception If an error occurs during the conversion process.
+     */
+    private RSAPrivateKey convertToRSAPrivateKey(String privateKeyString) throws Exception {
+        try {
+            privateKeyString = privateKeyString.replace("-----BEGIN PRIVATE KEY-----", "")
+                    .replace("-----END PRIVATE KEY-----", "")
+                    .replaceAll("\\s", "");
+
+            byte[] privateKeyBytes = Base64.getDecoder().decode(privateKeyString);
+            PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(privateKeyBytes);
+            KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+            return (RSAPrivateKey) keyFactory.generatePrivate(keySpec);
+        } catch (NoSuchAlgorithmException | InvalidKeySpecException | IllegalArgumentException e) {
+            throw new Exception("Error converting private key: " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Converts the given public key string to an RSAPublicKey object.
+     *
+     * @param publicKeyString the public key string to be converted
+     * @return the RSAPublicKey object converted from the public key string
+     * @throws Exception if there is an error during the conversion process
+     */
+    private RSAPublicKey convertToRSAPublicKey(String publicKeyString) throws Exception {
+        publicKeyString = publicKeyString.replace("-----BEGIN PUBLIC KEY-----", "")
+                .replace("-----END PUBLIC KEY-----", "")
+                .replaceAll("\\s", "");
+
+        byte[] publicKeyBytes = Base64.getDecoder().decode(publicKeyString);
+        X509EncodedKeySpec keySpec = new X509EncodedKeySpec(publicKeyBytes);
+        KeyFactory keyFactory = KeyFactory.getInstance("RSA");
```

Comment:
Is there a use-case where the KeyFactory object required would be different between two subsequent calls of this `convertToRSAPublicKey` method?
If not, then we can try to a single instance?

Reply:
don't think so it can be a different one, let me try creating a static instance

Reply:
Hi @debanjanc01 can we initialise the variable inside the static block like 
```
private static KeyFactory keyFactory;

    // Static block
static {
    try {
        keyFactory = KeyFactory.getInstance("RSA");
    } catch (NoSuchAlgorithmException e) {
        // Handle the exception
        e.printStackTrace();
    }
}
```

Reply:
Avoid mentioning the keyFactory as static since the class is already being initialised. You can just keep it as a private field/attribute of the class.
As for the initialisation of the KeyFactory, you can do it directly in the constructor, or inside a init() method annotated with `PostConstruct`